third_party/move/move-vm/runtime/src/runtime.rs

```diff
@@ -48,12 +50,20 @@ impl Clone for VMRuntime {
 impl VMRuntime {
     pub(crate) fn new(
         natives: impl IntoIterator<Item = (AccountAddress, Identifier, Identifier, NativeFunction)>,
+        deserializer_config: DeserializerConfig,
+        verifier_config: VerifierConfig,
         vm_config: VMConfig,
-    ) -> PartialVMResult<Self> {
-        Ok(VMRuntime {
-            loader: Loader::new(NativeFunctions::new(natives)?, vm_config),
+    ) -> Self {
+        VMRuntime {
+            loader: Loader::new(
+                NativeFunctions::new(natives)
```

Comment:
I mean the client can actually provide the hashmap, which will guarantee uniqueness... the only reason it is here is because we want to support iterators so we can use vectors.

What do you mean by into? 

So far my reasoning is:
1. We propagate a single error across the stack and do unwrap
2. We can just do unwrap and document APIs.

Do you think just using a hashmap would solve the problem?

Reply:
ah I think I was just mechanically following the code, and saw that what used to be an error was becoming a panic up until a certain point, must have just meant changing the signature of NativeFunctions::new, but I think that is worse than returning an Err an processing here, so nevermind.